sopel/modules/help.py

```diff
@@ -12,10 +12,81 @@
 
 import collections
 import requests
+import socket
 import textwrap
 
 from sopel.logger import get_logger
 from sopel.module import commands, rule, example, priority
+from sopel.config.types import (
+    StaticSection, ChoiceAttribute
+)
+
+
+# Pastebin handlers
+
+
+def post_to_clbin(bot, msg):
+    result = requests.post('https://clbin.com/', data={'clbin': msg})
+    result = result.text
+
+    if 'https://clbin.com/' in result:
+        return result
+    else:
+        LOGGER.error("Invalid result %s", result)
+        raise PostingException()
+
+
+def post_to_0x0(bot, msg):
+    payload = {'file': msg}
+    result = requests.post('https://0x0.st', files=payload)
+    return result.text
+
+
+def post_to_hastebin(bot, msg):
+    result = requests.post('https://hastebin.com/documents', data=msg)
+    result = result.json()
+    if 'key' not in result:
+        LOGGER.error("Invalid result %s", result)
+        raise PostingException()
+    return "https://hastebin.com/" + result['key']
+
+
+def post_to_termbin(bot, msg):
```

Comment:
This line in the file also needs to be updated:
https://github.com/sopel-irc/sopel/blob/38e30e67ad8dc5c90da87591d46f1f2f6167833e/sopel/modules/help.py#L169

If the socket cannot connect or is outright refused, it will not throw a `requests` error, but a [`socket.error`](https://docs.python.org/2/library/socket.html#socket.error)

The other (and IMO better) option is to just wrap the entire socket block (all socket related commands) in a `try`/`except`, catch and `socket` related exception, and raise `PostingError` instead. This may be best because there are at least four different non-nested `socket` error types that would otherwise have to be declared here individually; see next comment.